python_modules/dagster/dagster/_config/structured_config.py

```diff
@@ -319,7 +331,9 @@ def _safe_is_subclass(cls: Any, possible_parent_cls: Type) -> bool:
 
 
 def infer_schema_from_config_class(
-    model_cls: Type[Config], description: Optional[str] = None
+    model_cls: Type[Config],
+    description: Optional[str] = None,
+    wrap_config_type_fn: Optional[Callable[[ConfigType], ConfigType]] = None,
```

Comment:
passing in a callback that gets invoked at the end of this function feels unnecessarily clever. would prefer to see an approach where callsites manage this, even if it involves a little code duplication